e2e-tests/demand-backup-encrypted/run

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+check_pvc_md5() {
+    desc 'check backup file md5sum'
+    apply_config "$test_dir/conf/client.yml"
+    sleep 10
+    bak_client_pod=$(
+        kubectl_bin get pods \
+            --selector=name=backup-client \
+            -o 'jsonpath={.items[].metadata.name}'
+    )
+    wait_pod $bak_client_pod
+    kubectl_bin exec $bak_client_pod -- \
+        bash -c "cd /backup; md5sum -c md5sum.txt"
+    kubectl_bin delete \
+        -f $test_dir/conf/client.yml
+}
+
+run_backup() {
+    local cluster=$1
+    local backup1=$2
+
+    desc 'make backup'
+    kubectl_bin apply \
+        -f $test_dir/conf/$backup1.yml
+    wait_backup $backup1
+}
+
+run_recovery_check() {
+    local cluster=$1
+    local backup1=$2
+
+    desc 'recover backup'
+    kubectl_bin apply -f "$test_dir/conf/restore-${backup1}.yaml"
+    wait_backup_restore ${backup1}
+    kubectl_bin logs job/restore-job-${backup1}-${cluster}
+    kubectl_bin delete -f "$test_dir/conf/restore-${backup1}.yaml"
+    wait_for_running "$cluster-proxysql" 1
+    wait_for_running "$cluster-pxc" 3
+
+    sleep 20
+    desc 'check data after backup'
+    compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-0.$cluster-pxc -uroot -proot_password"
+    compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-1.$cluster-pxc -uroot -proot_password"
+    compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-2.$cluster-pxc -uroot -proot_password"
+
+    if [ "$backup1" != "on-demand-backup-minio" ]; then
+        desc 'copy backup'
+        bash $src_dir/deploy/backup/copy-backup.sh $backup1 $tmp_dir/backup
+    fi
+}
```

Comment:
can we move these functions inside `functions` file?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/354/commits/89fa96246eb421ce2d3a604cd3a19d4258d2a2dd